File: Dataset/Leetcode/train/125/88.cpp
 class Solution {
public:
    char isAbcorNumber(char s){
        //判断是不是字母或数字 
        if('A'<=s&&s<='Z'){
            //转化成小写
            s+=32;
            //Z与a之间隔了6位符号
            return s;
        }else if(('0'<=s&&s<='9')||('a'<=s&&s<='z')){
            return s;
        }
        return '#';
    }
    bool XXX(string s) {
        int start=0;
        int end=s.length()-1;
        while(start<end){
            char c1=isAbcorNumber(s[start]);
            char c2=isAbcorNumber(s[end]);
            while(start<end&&c1=='#'){
                start++;
                c1=isAbcorNumber(s[start]);
            }
            while(start<end&&c2=='#'){
                end--;
                c2=isAbcorNumber(s[end]);
            }
            if(c1!=c2){
                return false;
            }
            else{
                start++;
                end--;
            }
        }
        return true;
    }
};

File: Dataset/Leetcode/train/12/88.java
 class Solution {
    public String XXX(int num) {
        Integer[] arr = new Integer[]{1, 4,5, 9, 10, 40, 50, 90, 100, 400, 500, 900, 1000};
        String[] arr2 = new String[]{"I", "IV","V" ,"IX", "X", "XL", "L", "XC", "C", "CD", "D", "CM", "M"};
        StringBuilder sb = new StringBuilder();
        for (int i = arr.length - 1; i >= 0; i--) {
            int count = num / arr[i];
            int min=count * arr[i];
            num = min!=0?num % min:num;
            for (int j = 0; j < count; j++) {
                sb.append(arr2[i]);
            }
        }
        return sb.toString();
    }
}

